internal/k3s/install.go

```diff
@@ -0,0 +1,238 @@
+// if debug flag is not enabled
+func cleanup(debug bool) {
+	if !debug {
+		logger.Info().Msg("Cleaning up installation")
+
+		exec.Command("k3s-uninstall.sh").Run()
+		os.RemoveAll(k3sImagesPath)
+		os.Remove(k3sBinary())
+		os.Remove(k3sResolvConfPath)
+	}
+}
+
+func copyK3S() bundle.TarCallback {
+	return bundle.TarCallback{
+		FileName: "k3s",
+
+		Callback: func(ctx context.Context, _ fs.FileInfo, r io.Reader) (err error) {
+			logger.Info().Msg("Copying k3s binary")
+
+			f, err := os.OpenFile(k3sBinary(), os.O_CREATE|os.O_WRONLY, fs.FileMode(0755))
+			if err != nil {
+				return err
+			}
+			defer f.Close()
+
+			_, err = io.Copy(f, r)
+			if err != nil {
+				f.Close()
+				os.Remove(k3sBinary())
+				return err
+			}
+
+			return nil
+		},
+	}
+}
+
+func copyAirgapImages() bundle.TarCallback {
+	return bundle.TarCallback{
+		FileName: "k3s-airgap-*.tar*",
+
+		Callback: func(ctx context.Context, info fs.FileInfo, r io.Reader) (err error) {
+			logger.Info().Msg("Copying airgap images")
+
+			os.MkdirAll(k3sImagesPath, 0644)
+
+			var f *os.File
+			f, err = os.OpenFile(path.Join(k3sImagesPath, info.Name()), os.O_CREATE|os.O_WRONLY, fs.FileMode(0644))
+			if err != nil {
+				return err
+			}
+			defer f.Close()
+
+			_, err = io.Copy(f, r)
+			if err != nil {
+				err = errors.Join(err, os.Remove(path.Join(k3sImagesPath, info.Name())))
+				return err
+			}
+
+			return nil
+		},
+	}
+}
+
+func runInstallScript(c InstallConfig) bundle.TarCallback {
+	return bundle.TarCallback{
+		FileName: "install.sh",
+
+		Callback: func(ctx context.Context, fi fs.FileInfo, r io.Reader) error {
+			logger.Info().Msg("Starting k3s install")
+
+			if c.Hostname != "" {
+				os.Setenv("K3S_HOSTNAME", c.Hostname)
+				os.Setenv("K3S_NODE_NAME", c.Hostname)
+			}
+			overriden, err := resolvConfOverriden()
+			if err != nil {
+				return err
+			}
+			if overriden {
+				os.Setenv("K3S_RESOLV_CONF", k3sResolvConfPath)
+			}
+			os.Setenv("INSTALL_K3S_BIN_DIR", bundle.BundleBinDir())
+			os.Setenv("INSTALL_K3S_SKIP_DOWNLOAD", "true")
+			os.Setenv("INSTALL_K3S_SELINUX_WARN", "true")
+			os.Setenv("INSTALL_K3S_SKIP_SELINUX_RPM", "true")
+			os.Setenv("INSTALL_K3S_EXEC", c.k3sInstallArgs())
+
+			cmd := exec.CommandContext(ctx, "sh", "-")
+			cmd.Stdin = r
+
+			stdout, err := cmd.StdoutPipe()
+			if err != nil {
+				return err
+			}
+			stderr, err := cmd.StderrPipe()
+			if err != nil {
+				return err
+			}
+
+			err = cmd.Start()
+			if err != nil {
+				return err
+			}
+
+			go logReader(stdout, utils.InfoLevel)
+			go logReader(stderr, utils.InfoLevel)
+
+			err = cmd.Wait()
+			if err != nil {
+				return fmt.Errorf("install.sh: %w", errors.Join(err, ctx.Err()))
+			}
+
+			logger.Info().Msg("Install completed")
+
+			return nil
+		},
+	}
+}
+
+var logRegexp = regexp.MustCompile(`(\[(.+?)\]\s*)?(.+)`)
+
+// logReader parses log files from reader and uses our logging system
+func logReader(r io.Reader, lvl zerolog.Level) {
```

Comment:
I'd very much appreciate moving that into utils package because I need that for helm installation too